### PR TITLE
Remove zip usage in serializers

### PR DESCRIPTION
**Pull request by dcramer**
This simplifies some code/memory overhead.

@getsentry/api @getsentry/infrastructure

### CODE REVIEW

File: src/sentry/api/serializers/base.py [hunk 1]
```diff
@@ -12,7 +12,9 @@ def serialize(objects, user=None, serializer=None):
 
     if not objects:
         return objects
-    elif not isinstance(objects, (list, tuple)):
+    # sets aren't predictable, so generally you should use a list, but it's
+    # supported out of convenience
+    elif not isinstance(objects, (list, tuple, set, frozenset)):
```
Comment (mitsuhiko) [inline src/sentry/api/serializers/base.py]: I know we're not doing that for dicts either currently, but I had some good results in the past with paying the extra price to sort sets and dicts.  Makes them have a consistent order between machines and thus can be cached. Just a thing to note.

Comment (dcramer) [inline src/sentry/api/serializers/base.py]: only added this so I could pass them in the areas where we don't care about order here -- didn't feel worth it to coerce them all to lists just to pass them into this func

**Comment by mitsuhiko** [comment]
Looks like this depends on breadcrumbs landing first.

File: src/sentry/api/serializers/models/project.py [hunk 2]
```diff
@@ -42,7 +42,7 @@ def serialize(self, obj, attrs, user):
         from sentry import features
 
         feature_list = []
-        for feature in ('event-types', 'global-events', 'user-reports', 'dsym'):
+        for feature in ('breadcrumbs', 'event-types', 'global-events', 'user-reports', 'dsym'):
```
Comment (dcramer) [inline src/sentry/api/serializers/models/project.py]: oops, yeah wasnt intending for this to end up here

i'll back this out of this PR and stuff it into the other

**Comment by mitsuhiko** [comment]
lgtm

**Pull request merged**